packages/desktop-client/src/hooks/useCategories.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+
+import { useActions } from './useActions';
+
+export default function useCategories() {
+  const { getCategories } = useActions();
+
+  const categories = useSelector(state => state.queries.categories.list);
+  const categoriesGrouped = useSelector(
+    state => state.queries.categories.grouped,
+  );
+
+  useEffect(() => {
+    if (categories.length === 0) {
+      getCategories();
+    }
+  }, []);
+
+  return {
+    list: categories,
+    grouped: categoriesGrouped,
+  };
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏small improvement to re-render less times

```suggestion
  return useSelector(state => state.queries.categories);
```